### ISSUE TITLE
DnD is triggered instead of silence

### ISSUE THREAD

**Issue by roninvanderwerke** [keymapperorg/KeyMapper#1133]
**Developer TODO (don't remove)**
- [ ] create new branch. put issue number at start of name if not a very quick fix.
- [ ] write tests. put issue number in comment
- [ ] update documentation
- [ ] merge and delete branch (don't squash because want commit history to see why I made changes)

**Discord message link/email recipient**
roninvdwerke2000@gmail.com

**Describe the bug**

The action "Ringer mode - Silent" triggers "DnD mode" with either cycling or as a specific action.

**Expected behavior**
It should change the ringer mode to silence not activate DnD.

**Smartphone (please complete the following information):**
 - Device: Google Pixel 7
 - Android version: 13

**Comment by sds100** [comment]
Related to #762

**Comment by sds100** [comment]
TLDR: I can't fix this :(

I spent a couple hours trying to find a work around for this and it seems like it is intentional behavior in Android. I dove into the AudioService class in AOSP and it seems like the system enables DND when an "external"/3rd party app requests to turn on silent ringer mode.

https://issuetracker.google.com/issues/237819541
https://stackoverflow.com/questions/58044974/enable-silent-mode-in-android-without-triggering-do-not-disturb
https://stackoverflow.com/questions/57832788/silent-phone-without-dont-disturb

**Comment by sds100** [comment]
This was actually fixed in the end by #762 by using PRO mode

**Comment by Altarentys** [comment]
> This was actually fixed in the end by [#762](https://github.com/keymapperorg/KeyMapper/issues/762) by using PRO mode

Hi! Pro mode is referring to "Require WRITE_SECURE_SETTINGS permission" option that must be enabled through adb, right?

If yes, I did enabled it through adb, but still the action "Change ringer mode : Silent" activates DND instead of silent.

I have a Pixel 5 with Android 14.

https://github.com/user-attachments/assets/71fd2d3b-1ad4-4824-9a77-f17ae84beaea
